src/app/modules/submission/submission.utils.ts

```diff
@@ -68,3 +92,237 @@ export const extractEmailConfirmationData = (
+    return responses.reduce<FieldIdSet>((acc, response) => {
+      const responseId = response._id
+      if (visibilityPredicate(response)) {
+        acc.add(responseId)
+      }
+      return acc
+    }, new Set()) as VisibleResponseIdSet
+  }
+
+  /**
+   * Generates a set of response IDs that are verifiable. A response is
+   * said to be verifiable if its corresponding form field has the
+   * isVerifiable property set to `true`.
+   * @param responses
+   * @param fieldMap
+   * @returns verifiableResponseIds - Generated set of response IDs.
+   * @protected
+   */
+  protected static getVerifiableResponseIds(
+    responses: FieldResponse[],
+    fieldMap: ValidatedFieldMap,
+  ): VerifiableResponseIdSet {
+    return responses.reduce<FieldIdSet>((acc, response) => {
+      const responseId = response._id
+      const formField = fieldMap[responseId]
+      if (formField.isVerifiable) {
+        acc.add(responseId)
+      }
+      return acc
+    }, new Set()) as VerifiableResponseIdSet
+  }
+
+  /**
+   * Prior to the introduction of IncomingResponse, responses were
+   * stored in a ProcessedFieldResponse[]. This method helps to
+   * create a ProcessedFieldResponse object that can be used in
+   * functions that have not been refactored to take in the new
+   * IncomingResponse object.
+   *
+   * A ProcessedResponse is just a regular FieldResponse with some
+   * metadata inserted into them. This function takes in several
+   * data sets that help determine what each metadata field should
+   * contain.
+   * @param response
+   * @param fieldMap
+   * @param visibleResponseIds
+   * @param verifiableResponseIds
+   * @returns processedFieldResponse - The ProcessedFieldResponse
+   * @protected
+   */
+  protected static getLegacyProcessedFieldResponse(
+    response: FieldResponse,
+    fieldMap: ValidatedFieldMap,
+    visibleResponseIds: VisibleResponseIdSet,
+    verifiableResponseIds: VerifiableResponseIdSet,
+  ): ProcessedFieldResponse {
+    const responseId = response._id
+    const formField = fieldMap[responseId]
+    const processedResponse: ProcessedFieldResponse = {
+      ...response,
+      isVisible: visibleResponseIds.has(responseId),
+      question: formField.getQuestion(),
+    }
+
+    /**
+     * This pattern of mutating processedResponse is copied from the original
+     * getProcessedResponse function. TODO: Check if can just move this into
+     * the object instantiation stage, preserving the possible undefined value.
+     */
+    if (verifiableResponseIds.has(responseId)) {
+      processedResponse.isUserVerified = true
+    }
+
+    return processedResponse
+  }
+
+  protected static validateResponses(
+    responses: FieldResponse[],
+    form: IFormDocument,
+    fieldMap: ValidatedFieldMap,
+    visibleFieldIds: FieldIdSet,
+    visibleResponseIds: VisibleResponseIdSet,
+    verifiableResponseIds: VerifiableResponseIdSet,
+  ): Result<true, ProcessingError | ValidateFieldError> {
```

Comment:
suggest making this an object instead of positional arguments as discussed on slack